neuraxle/base.py

```diff
@@ -945,24 +949,38 @@ def inverse_transform(self, processed_outputs):
         Inverse Transform the given transformed data inputs.
 
         :func:`~neuraxle.base.BaseStep.mutate` or :func:`~neuraxle.base.BaseStep.reverse` can be called to change the default transform behavior :
-        ```
-            class MultiplyBy(NonFittableMixin, BaseStep):
+
+        .. code-block:: python
+            class MultiplyByN(NonFittableMixin, BaseStep):
+                def __init__(self, multiply_by):
+                    NonFittableMixin.__init__(self)
+                    BaseStep.__init__(
+                        self,
+                        hyperparams=HyperparameterSamples({
+                            'multiply_by': multiply_by
+                        })
+                    )
+
                 def transform(self, data_inputs):
                     return data_inputs * self.hyperparams['multiply_by']
 
                 def inverse_transform(self, processed_outputs):
                     return data_inputs / self.hyperparams['multiply_by']
 
-            p = Pipeline([MultiplyBy()])
+            p = Pipeline([
+                MultiplyByN(multiply_by=2)
+            ])
+
+            outputs = p.transform(np.array[1, 2])
+
+            assert np.array_equal(outputs, np.array([2, 4]))
 
             p = p.mutate(new_method="inverse_transform", method_to_assign_to="transform")
-            p.transform(processed_outputs) # this will call inverse transform method instead
+            # or p = reverse(p)
 
-            # is equivalent to :
+            outputs = p.transform(processed_outputs) # this will call inverse transform method instead
 
-            p = reverse(p)
-            p.transform(processed_outputs) # this will call inverse transform method instead
-        ```
+            assert np.array_equal(outputs, np.array([1, 2]))
```

Comment:
This is what I think I asked for: 
```
_out = p.transform(_in)
_regenerated_in = reversed(p).transform(_out)
assert _regenerated_in == _in
```

Reply:
You could as well precise that the inverse_transform method is optional to implement